src/main/java/edu/iis/powp/command/factory/CommandFactoryWindowController.java

```diff
@@ -0,0 +1,133 @@
+package edu.iis.powp.command.factory;
+
+import edu.iis.powp.command.ICompoundCommand;
+import edu.iis.powp.command.IPlotterCommand;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.io.Serializable;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Parameter;
+import java.util.*;
+import java.util.List;
+
+class CommandFactoryWindowController implements Serializable {
+
+    private final CommandFactoryWindow commandFactoryWindow;
+
+    CommandFactoryWindowController(CommandFactoryWindow commandFactoryWindow) {
+        this.commandFactoryWindow = commandFactoryWindow;
+    }
+
+    ActionListener handleClearButton(DefaultListModel<IPlotterCommand> jListModel) {
+        return new ActionListener() {
+
+            @Override public void actionPerformed(final ActionEvent e) {
```

Comment:
Better, but: A controller made up from ActionListeners seems awkward. It would be better to equip it with purely logic methods, extract an interface and define ActionListeners inside GUI class